amundsen_application/static/js/components/SearchPage/SearchFilter/index.tsx

```diff
@@ -1,168 +1,153 @@
-            count: 100, // pull value from state
-          },
-          {
-            value: 'hive', // value.id
-            labelText: 'Hive', // value.displayName
-            checked: true, // pull value or infer value from state
-            count: 100, // pull value from state
-          },
-          {
-            value: 'druid', // value.id
-            labelText: 'Druid', // value.displayName
-            checked: true, // pull value or infer value from state
-            count: 0, // pull value from state
-          },
-          {
-            value: 's3', // value.id
-            labelText: 'S3 Buckets', // value.displayName
-            checked: false, // pull value or infer value from state
-            count: 100, // pull value from state
-          }
-        ]
-      },
-      {
-        title: 'Badges', // category.displayName
-        categoryId: 'badges', // category.id
-        inputProperties: [
-          {
-            value: 'sla', // value.id
-            labelText: 'Missed SLA', // value.displayName
-            checked: true, // pull value or infer value from state
-            count: 3, // pull value from state
-          },
-          {
-            value: 'quality', // value.id
-            labelText: 'High Quality', // value.displayName
-            checked: true, // pull value or infer value from state
-            count: 12, // pull value from state
-          },
-          {
-            value: 'pii', // value.id
-            labelText: 'PII', // value.displayName
-            checked: false, // pull value or infer value from state
-            count: 34, // pull value from state
-          },
-          {
-            value: 'deprecated', // value.id
-            labelText: 'Deprecated', // value.displayName
-            checked: false, // pull value or infer value from state
-            count: 3, // pull value from state
-          }
-        ]
+  const resourceType = state.search.selectedTab;
+  const filterCategories = getFilterConfigByResource(resourceType);
+  const filterState = state.search.filters;
+
+  const checkBoxSections = [];
+  const inputSections = [];
+
+  let currentFilterValue;
+  if (filterCategories) {
+    /* checkbox sections */
+    filterCategories.forEach((categoryConfig) => {
+      currentFilterValue = filterState[resourceType][categoryConfig.value];
+      if (categoryConfig.type === FilterType.MULTI_SELECT_VALUE) {
+        checkBoxSections.push({
+          title: categoryConfig.displayName,
+          categoryId: categoryConfig.value,
+          helpText: categoryConfig.helpText,
+          properties: categoryConfig.options.map((option) => {
+            return {
+              value: option.value,
+              labelText: option.displayName,
+              /* tertiary statement allows for 'false' to be explicity set as opposed to 'undefined' */
+              checked: currentFilterValue && currentFilterValue[option.value] ? true : false,
```

Comment:
I think I wanted to avoid connecting all of `SearchFilter` children to Redux, but eventually had to do that any to dispatch actions. I will look into moving the appropriate pieces of logic into `CheckboxFilter`, `FilterSection`, and `InputFilter`.

Reply:
I think there are pros and cons of both approaches (passing down props to children vs connecting children). I think the former approach works best for simple use cases like passing down a single value or event handler to a direct descendant. Once you start passing down complex values and with state changing functions and multiple levels of nesting, connecting each child component becomes the simpler solution (not sure about performance).

Reply:
@danwom what are your thoughts on the modification given that the following changes would be required:
1. `SearchFilter` will still have to loop through each of the `filterCategories` on the application configuration. This is so that it knows how many `FilterSection` to create. 
2. `FilterSection` would then get the value for its `categoryId` from the redux state, and would then -- like `SearchFilter` -- has branching logic for each filter type where is calculates `hasValue` for the purposed of knowing if it needs to render a `< Clear` button. For checkboxes this requires looping through each option and checking if it exists in the state.
3. `CheckboxFilter` will duplicate the same work done `FilterSection` of looping through each option and checking if it exists in the state in order to figure out which boxes should be checked. 

I'm not confident that the suggestion provides a clear benefit that is worth the work effort. Initial code explorations don't result in saving on any mappings or calculations, but rather spreading some of it out from `SearchFilter` into other places, and even duplicating logic in `FilterSection` and `CheckboxFilter` in the multi-select case.

Do you have any other thoughts that you think we should strongly consider when it comes to modifying this piece of the implementation vs not? Also let me know if there was just something about your suggestion that I actually misunderstood.

Reply:
1. Looping through the list of `filterCategories` is a given since we allow for a flexible configuration. If we had a pre-defined list of categories that could either be enabled or disabled, we could hard-code those into the template. This would limit the flexibility of filtering different fields.
2. `FilterSection` wouldn't need to get its `categoryId` from redux as the section should receive the config object as a prop.

In SearchFilter
```
{ filterCategories.map(filterConfig =>  <FilterSection config={filterConfig} resourceType={resourceType}/>) } 
```

In FilterSection
```
render() {
  let content;
  if (this.props.config.type === FilterType.MULTI_SELECT_VALUE) {
    content = this.props.config.options.map(option => {
      return <CheckboxOption value={option.value} label={option.label} />
    }) 
  } else {
    content <InputFilter .../>
  }
  return (
    ...
    { label }
    { this.props.hasValue && <button>clear</button> } 
    { content }
    ...
   )
  }
}

mapStateToProps(globalState) {
  const filterState = state.search.filters;
  const filterValue = filterState[resourceType][categoryConfig.value];
  let hasValue = false;
  if (categoryConfig.type === FilterType.MULTI_SELECT_VALUE) {
    filterValue.keys().forEach(key => {
      if (filterValue[key] === true) { hasValue = true; }
    })
  } else {
    hasValue = !!filterValue;
  }
}
```

Reply:
I think I see it now. The part of what you were suggesting that I was missing was that `SearchFilter` would no longer render both `FilterSection` and its child.

I also see another benefit that makes this go beyond just a cleanup. With what you suggested users will have full control over the order of their filters in the UI based on the order of their `filterCategories` array. Whereas the current implementation forces filters with the same type to be rendered together.

With this current suggestion though I think about what will happen when we have multi select values that render with something else besides a checkbox, like a tag input type UI. Do you think it's worth renaming the filter types according to desired UI like `FilterType.INPUT_SELECT` and `FilterType.CHECKBOX_SELECT`? This could be reasonable because there are a limited number of best practice filter UIs:
- radio select, as another single option choice
- tag input-like UI ,as another multi option choice
- slider bar for a range (some users have requested searching by last_updated) 
- ...might be more but I feel confident that there are a limited number of reasonable ones

This way the filter type can be better used to let us know which UI to use, and that also implicitly lets us know if it supports multiple options or not based on that type.

Reply:
Yes, the proposed change would eliminate the need to group based on type as an added bonus. In practice it's unclear how much the order actually matters. 

I think switching from single/multi to an explicit input type is a good change and allows for additional input types in the future.

Reply:
Cool thanks changing things up according to this discussion and will raise any issues -- hopefully none.

Order might not matter to us right now but it might matter to others. I notice that filter UIs with many options like Amazon seem to make an intentional choice on order based on an understanding of their users or what would best narrow results. (the checkbox for "prime"/one-day delivery, rating, brand and cost are all at the top, while something like who exactly sells it is all the way at the bottom)

Reply:
Recent commit contains these updates -- changes were made with a focus on each search component becoming responsible for getting any reducer-dependent state props from the application state.